homeassistant/components/forecast_solar/config_flow.py

```diff
@@ -84,12 +85,26 @@ def __init__(self, config_entry: ConfigEntry) -> None:
         """Initialize options flow."""
         self.config_entry = config_entry
 
+    async def validate_input(self, user_input: dict[str, Any]) -> str | None:
+        """Validate the input from the config options flow."""
+        if user_input.get(CONF_API_KEY):
+            api_key = user_input[CONF_API_KEY]
+            if not re.match("[a-zA-Z0-9]{16}", api_key):
+                raise Exception("invalid_api_key")
+        return None
+
     async def async_step_init(
         self, user_input: dict[str, Any] | None = None
     ) -> FlowResult:
         """Manage the options."""
+        errors = {}
         if user_input is not None:
-            return self.async_create_entry(title="", data=user_input)
+            try:
+                await self.validate_input(user_input)
```

Comment:
Instead of making a method that does a validation, add it as an actual validator to the data_schema

Reply:
nvm that is not serializable.